service/controller/resource/nodes/status.go

```diff
@@ -32,13 +36,15 @@ func (r *Resource) GetResourceStatus(customObject providerv1alpha1.AzureConfig,
 }
 
 func (r *Resource) SetResourceStatus(customObject providerv1alpha1.AzureConfig, t string, s string) error {
+	ctx := context.Background()
```

Comment:
We should add context as a parameter to the `SetResourceStatus` func and then propagate it from the caller.